firmware/src/rdm_responder.c

```diff
@@ -51,18 +103,19 @@ void RDMResponder_ResetToFactoryDefaults() {
   g_responder.dmx_start_address = INVALID_DMX_START_ADDRESS;
   g_responder.dmx_footprint = 0;
   g_responder.sub_device_count = 0;
-  g_responder.sensor_count = 0;
   g_responder.current_personality = 0;
   g_responder.personality_count = 0;
   g_responder.is_muted = false;
   g_responder.identify_on = false;
+  g_responder.sensors = NULL;
 
   if (g_responder.def) {
-    strncpy(g_responder.device_label,
-            g_responder.def->default_device_label,
-            RDMUtil_SafeStringLength(g_responder.def->default_device_label,
-                                     RDM_DEFAULT_STRING_SIZE));
-    g_responder.device_label[RDM_DEFAULT_STRING_SIZE] = 0;
+    unsigned int len = RDMUtil_StringCopy(
+        g_responder.device_label,
+        RDM_DEFAULT_STRING_SIZE,
+        g_responder.def->default_device_label,
+        RDM_DEFAULT_STRING_SIZE);
+    g_responder.device_label[len] = 0;
```

Comment:
It looks like this adds the null in RDMUtil_StringCopy so you don't need to now?